packages/cli-kit/src/public/node/terminal-layout.ts

```diff
@@ -0,0 +1,64 @@
+const MIN_TERMINAL_WIDTH = 80
+const TERMINAL_WIDTH_RATIO = 2 / 3
+const TRUNCATION_SUFFIX_LENGTH = 3
+const CURSOR_UP_AND_CLEAR = '\x1b[1A\x1b[2K\r'
+
+/**
+ * Calculates the length of a string, ignoring ANSI escape codes.
+ * (That lets use calculate how much space the string will actually take up when displayed.).
+ *
+ * @param text - The text to measure.
+ * @returns The length of the text without ANSI escape codes.
+ */
+function getVisualLength(text: string): number {
+  // eslint-disable-next-line no-control-regex
+  return text.replace(/\x1b\[[0-9;]*m/g, '').length
+}
+
+/**
+ * Creates right-aligned text by padding between left and right text.
+ *
+ * @param leftText - The text to display on the left.
+ * @param rightText - The text to display on the right.
+ * @returns The formatted string with proper alignment.
+ */
+export function createRightAlignedText(leftText: string, rightText: string): string {
```

Comment:
I got a bit confused by this function name. Do you mean right is sense "proper" and not in the sense of direction, right? ;)

I don't know maybe calling it something like `spreadText` would make it more straightforward.

Reply:
"Right-aligned text" is the official name of this concept, I think naming it anything else would be more confusing -- "spread text" could mean a lot of things, but "right-aligned" means precisely lining it up to the right edge, which is what we're doing. So I feel like this is the right name, but definitely open if anyone has other suggestions!

Reply:
I guess this function does also print left-aligned text too... let me think harder about the name for this.